solidity/test/InterchainAccountMirror.t.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.13;
+
+import {Test} from "forge-std/Test.sol";
+
+import {InterchainAccountMirrorCalldata} from "../contracts/middleware/libs/InterchainAccountMirrorCalldata.sol";
+import {InterchainAccountMirror} from "../contracts/middleware/InterchainAccountMirror.sol";
+import {InterchainAccountMirrorFactory} from "../contracts/middleware/InterchainAccountMirrorFactory.sol";
+import {InterchainAccountRouter} from "../contracts/middleware/InterchainAccountRouter.sol";
+import {CallLib} from "../contracts/middleware/libs/Call.sol";
+
+contract InterchainAccountMirrorTest is Test {
+    event InterchainAccountMirrorDeployed(
+        uint32 indexed destination,
+        bytes32 indexed target,
+        address indexed owner,
+        address mirror
+    );
+
+    address icaRouter = address(0x32);
+
+    InterchainAccountMirrorFactory icaMirrorFactory;
+
+    function setUp() public {
+        icaMirrorFactory = new InterchainAccountMirrorFactory(icaRouter);
+    }
+
+    function test(
+        address owner,
+        bytes calldata data,
+        uint256 value,
+        uint32 destination,
+        bytes32 target,
+        bytes32 destinationIcaRouter,
+        bytes32 destinationIsmAddress
+    ) public payable {
+        vm.expectEmit(true, true, true, false);
+        emit InterchainAccountMirrorDeployed(
+            destination,
+            target,
+            owner,
+            address(0x0) // assume CREATE2 works as expected
```

Comment:
I dont get this comment?

Reply:
im just not testing the CREATE2 derivation

Reply:
Still took me a second to understand that you are not asserting the last argument, maybe say something like "Not asserting the create2 derivation here" or something like that